### DISCUSSION TITLE
Suggestion: add "copy as curl" feature for diagnosing queries

### DISCUSSION THREAD

**Discussion by bboreham** [grafana/grafana#42079] [Observability]
In Chrome browser devtools there is a command "copy as cURL":
![image](https://user-images.githubusercontent.com/8125524/142871337-2fe8f8cd-cb3f-40cc-b191-b7c48515fea8.png)

this enables a request to be taken aside from the UI and repeated, which can be useful to narrow down the source of a problem.

My suggestion is to add a button like this to Grafana.

Mostly I use Prometheus data, so for me the request would be a Prometheus API request.
This used to be approximated by the Query Inspector in Grafana, but since the data source was moved to the Go back-end the Inspector shows only internal front-end/back-end conversations hence is not useful to diagnose the Prometheus query.

**Comment by gabor** [comment]
seems to be about solving the same problem as https://github.com/grafana/grafana/discussions/44275 , so marking as duplicate.

**Comment by bboreham** [comment]
The proposed solution is nothing like what I asked for.

**Reply by gabor** [reply]
i see, sorry for the misunderstanding, i removed the duplicate-mark.

**Reply by bboreham** [reply]
I agree the two requests are in the same general area.

My suggestion here is that Grafana generate the text of a curl command to hit the Prometheus API, which is structurally correct and complete.
To get there from the suggestion which closed #44275 you have know the exact format and encode all the parameters by hand.

**Reply by gabor** [reply]
one complication is sensitive-data and such.. for example, one may not want to reveal the auth-http-header's values to every grafana user. (it's not unsolvable, just something we will have to keep in mind)

**Reply by bboreham** [reply]
I would be OK with having the auth token redacted by Grafana so I had to add it back manually.
The main thing is to get "what exactly did Grafana send as the request to Prometheus?" without a lot of manual effort.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
